### PR TITLE
return a non-zero return code if no subtitles were found

### PR DESCRIPTION
**Pull request by anukul**
Previous PR: https://github.com/CCExtractor/ccextractor/pull/535

### CODE REVIEW

**Comment by cfsmp3** [comment]
Could work. @canihavesomecoffee what does the CI say?
@anukul I can't see a pending code-in task, did you claim and submit?

**Comment by anukul** [comment]
Yes, you had approved it tentatively. It is awaiting parental consent approval for completion.

**Comment by canihavesomecoffee** [comment]
@cfsmp3 The PR is fine, except that it'll break the sample platform until I can fix it.

**Comment by anukul** [comment]
Test 139 looks fine to me.

`➜  fix l
ccextractor  inp.mpg
➜  fix ./ccextractor inp.mpg 
CCExtractor 0.83, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: inp.mpg
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: No]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: inp.mpg
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
Creating inp.srt


New video information found
[1920 * 1080] [AR: 03 - 16:9] [FR: 04 - 29.97] [progressive: no]

100%  |  00:19
Number of NAL_type_7: 0
Number of VCL_HRD: 0
Number of NAL HRD: 0
Number of jump-in-frames: 0
CAPTION STATUS IS 1ected_sei_length: 0


Total frames time:	  00:00:20:020  (600 frames at 29.97fps)

Min PTS:				00:00:00:400
Max PTS:				00:00:20:453
Length:				 00:00:20:053

Initial GOP time:	   00:00:00:000
Final GOP time:		 00:00:19:500+15F
Diff. GOP length:	   00:00:19:500+15F	(00:00:20:000)


Total user data fields: 600
HDTV type user data fields: 600
Done, processing time = 2 seconds
This is beta software. Report issues to carlos at ccextractor org...
➜  fix wc -l inp.srt 
15 inp.srt
➜  fix 
`

**Comment by anukul** [comment]
Okay, my workaround for mp4 is incorrect. I had assumed that `ctx->freport.mp4_cc_track_cnt` would store a non zero value in the case when captions are present. But I guess it's not so.

Should I modify the gpacmp4 libary?

**Comment by cfsmp3** [comment]
No, that doesn't help. You can have a CC track but no subtitles inside of
it but any reason, for example. Only way is delay opening the output file
until the first write.

**Comment by anukul** [comment]
The last commit fixes Tests 28, 29, 30.
I'm not completely convinced as to why Test 139 should be returning 10, it does generate a non-empty subtitle file.

**Comment by anukul** [comment]
@cfsmp3 @canihavesomecoffee

File: src/ccextractor.c [hunk 1]
```diff
@@ -215,27 +214,33 @@ int main(int argc, char *argv[])
...
+				if (!ret) ret = tmp;
 				break;
 			case CCX_SM_MP4:
 				mprint ("\rAnalyzing data with GPAC (MP4 library)\n");
 				close_input_file(ctx); // No need to have it open. GPAC will do it for us
 				processmp4(ctx, &ctx->mp4_cfg, ctx->inputfile[ctx->current_file]);
 				if (ccx_options.print_file_reports)
 					print_file_report(ctx);
+				tmp = ctx->freport.mp4_cc_track_cnt;
+				if (!ret) ret = tmp;
```
Comment (anshul1912) [inline src/ccextractor.c]: why tmp variable needed, why cant you just capture the return value in ret variable

Comment (anukul) [inline src/ccextractor.c]: because of [this](https://github.com/CCExtractor/ccextractor/pull/535#issuecomment-267477706)

Comment (anukul) [inline src/ccextractor.c]: because of (this)[https://github.com/CCExtractor/ccextractor/pull/535#issuecomment-267771679].

Comment (anukul) [inline src/ccextractor.c]: because of [this](https://github.com/CCExtractor/ccextractor/pull/535#issuecomment-267771679).

**Pull request merged**